### ISSUE TITLE
CA1508 rule wrongfully warns that a params argument can't ever be null

### ISSUE THREAD

**Issue by Crono1981** [dotnet/roslyn-analyzers#4502]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

Code analysis assumes that an array parameter may never be passed as null. That is incorrect. If the method is invoked with an _array variable_ that itself is `null`, then the method will indeed receive a null array parameter.

```cs
// ...

public void MyMethod(params string[] args)
{
    if (args == null) throw new ArgumentNullException(nameof(args));
    // ...
}

// ...
string[] args = null;
MyMethod(args);
// ...
```

### Steps To Reproduce

1- Create any method that takes a parameter array;
2- Within the method, write an `if` statement that checks if the array instance itself (not the content) is `null`.

### Expected behavior

Code analysis doesn't raise any warnings.

### Actual behavior

Code analysis warns that `args == null` will always be `false`.

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/3845
